### DISCUSSION TITLE
Discussion: Updated MUXTestApp Logo to replace the current one?

### DISCUSSION THREAD

**Discussion by mdtauk** [microsoft/microsoft-ui-xaml#7471] [General]
# Discussion: [your title here] 

Should the current Logo used for the WinUI MUXTestApp get updated to something more fluent once the project is fully open sourced.

![image](https://user-images.githubusercontent.com/7389110/108767208-0bc08f80-754e-11eb-8e26-0907bd3fb7c4.png)

## Related Links

The Current logo is a bit of a Metro throwback...
![Square150x150Logo scale-200](https://user-images.githubusercontent.com/7389110/108767392-4c200d80-754e-11eb-9f7d-826a9075ed36.png)

**Comment by StephenLPeters** [comment]
Seems cool, would you like to contribute the change? :)

**Comment by marcelwgn** [comment]
Having a new icon for the test app would be amazing, @mdtauk do you already have something specific in mind? Maybe something that incorporates the WinUI logo?

If we have settled on the new logo, I would be more than happy to help create a PR to update the logo!

**Comment by mdtauk** [comment]
I was just toying around with one idea - using the green colour used by the Dev tag for Edge and other apps - but keeping the tool iconography.

Its not a high priority item for sure, and could be a community derived design.

**Comment by dphfox** [comment]
Sounds like a good idea. I think it could be more consistent with other fluent icons though - here's a quick mockup I made based on your idea:

![image](https://user-images.githubusercontent.com/11815374/108996746-acba6200-7696-11eb-84b7-7353ddbbf789.png)

![image](https://user-images.githubusercontent.com/11815374/108996693-9a402880-7696-11eb-927c-8316f2df9677.png)

**Comment by mdtauk** [comment]
> Sounds like a good idea. I think it could be more consistent with other fluent icons though - here's a quick mockup I made based on your idea:
> 
> ![image](https://user-images.githubusercontent.com/11815374/108996746-acba6200-7696-11eb-84b7-7353ddbbf789.png)
> 
> ![image](https://user-images.githubusercontent.com/11815374/108996693-9a402880-7696-11eb-927c-8316f2df9677.png)

This is not a public facing app, so it shouldn't match the others too closely IMO.
![image](https://user-images.githubusercontent.com/7389110/109041768-4bf74d80-76c7-11eb-82f7-1e4dac1cb27e.png)

The green Dev badge is used for in-development apps, so that is why I used it for my quick idea

**Comment by marcelwgn** [comment]
Speaking from a "user" point here, ideally, I would be able to know what MUXControlsTestApp is about because right now the logo and it's purpose have little connection. I really like @mdtauk idea with the "dev badge" green.

**Comment by mdtauk** [comment]
> Speaking from a "user" point here, ideally, I would be able to know what MUXControlsTestApp is about because right now the logo and it's purpose have little connection. I really like @mdtauk idea with the "dev badge" green.
>
Is there an established visual metaphor for tests or testing?  It is a developer app, and tests the UI and features of WinUI - so if you break it down, there is
- Development
- UI
- Testing

As an Icon...
- Green Dev Badge and Tool
- ?
- ?

**Comment by marcelwgn** [comment]
I'm not sure if there is iconology for testing really, Something that is used sometimes is some form of checkmark (or checklist) or test tube ([Reference](https://www.flaticon.com/free-icons/testing)), but not sure how good those look when incorporated into the logo.

**Comment by mdtauk** [comment]
![image](https://user-images.githubusercontent.com/7389110/109052396-79e28f00-76d3-11eb-9e0c-059658dedb22.png)

This adds more of the UI elements - but nothing for testing

A modern window with NavigationView, extended into titlebar area - the "material" with the acrylic like Wrench

**Comment by dphfox** [comment]
> ![image](https://user-images.githubusercontent.com/7389110/109052396-79e28f00-76d3-11eb-9e0c-059658dedb22.png)
> 
> This adds more of the UI elements - but nothing for testing
> 
> A modern window with NavigationView, extended into titlebar area - the "material" with the acrylic like Wrench

Thanks for the input! I don't think your icon would hold up at small sizes though - in particular, the use of the gradient and acrylic effect on the wrench, as well as the partial obscuring of the 'dev' badge, would likely cause readability issues at small sizes. 

My other criticism, though it's more my opinion than an actual usabilty problem, is that I don't agree with the switch to using a navigation pane to illustrate the app window. I prefer to emphasise the title bar rather than the navigation pane, since it's a universal component shared by all windows, and should generally be more recognisable and 'window-like'.

Based on the feedback in this thread, I've made some changes to my original design to try and improve it - notably, I've switched from the blue icon colour used by Windows apps to the grey icon colours from Windows Terminal, to give it more of a 'development' feel and to visually separate it from other apps on the system at a glance. I did like the 'dev badge' icon idea, so I've added it too.

How's this?
![image](https://user-images.githubusercontent.com/11815374/109064279-1e6bcd80-76e2-11eb-9d81-9b867c969b80.png)

![Frame 37](https://user-images.githubusercontent.com/11815374/109069094-a81e9980-76e8-11eb-98a1-b945a27d94ca.png)

**Comment by mdtauk** [comment]
> > ![image](https://user-images.githubusercontent.com/7389110/109052396-79e28f00-76d3-11eb-9e0c-059658dedb22.png)
> > This adds more of the UI elements - but nothing for testing
> > A modern window with NavigationView, extended into titlebar area - the "material" with the acrylic like Wrench
> 
> Thanks for the input! I don't think your icon would hold up at small sizes though - in particular, the use of the gradient and acrylic effect on the wrench, as well as the partial obscuring of the 'dev' badge, would likely cause readability issues at small sizes.
> 
The detail would be reduced as the icon size gets smaller of course, the added details come as you scale up.

![image](https://user-images.githubusercontent.com/7389110/109066888-a56e7500-76e5-11eb-9f72-96ea748bf79c.png)

![image](https://user-images.githubusercontent.com/7389110/109067457-7573a180-76e6-11eb-9b91-2df6e4298d78.png)

![image](https://user-images.githubusercontent.com/7389110/109067527-8ae8cb80-76e6-11eb-8015-3a008fa943c6.png)


> My other criticism, though it's more my opinion than an actual usabilty problem, is that I don't agree with the switch to using a navigation pane to illustrate the app window. I prefer to emphasise the title bar rather than the navigation pane, since it's a universal component shared by all windows, and should generally be more recognisable and 'window-like'.
> 
The titlebar as an element is not exactly modern, and most modern apps blend into it, so whilst I appreciate the window-like representation, but this app is less about the app, and more about testing UI - the NavigationView has become the key pattern for app layout and navigation.

**Comment by dphfox** [comment]
> The titlebar as an element is not exactly modern, and most modern apps blend into it, so whilst I appreciate the window-like representation, but this app is less about the app, and more about testing UI - the NavigationView has become the key pattern for app layout and navigation.

While I agree with your general logic, and do agree that NavigationViews are important, I still don't think it justifies the change.

In a desktop environment, the most common way of representing an interface in an icon is to represent it's window chrome. Users will generally be most familiar with this design, as it has been used across a majority of apps for the past few decades across multiple operating systems and desktop environments. It is important that we don't unnecessarily change widely accepted and used metaphors, otherwise it might be confusing or unintuitive to users.

While it's true that many apps don't visually distinguish their title bar from their main content area, the title bar portrayed in icons is still an important visual cue - it communicates to the user that the shape they're looking at represents a window. The title bar shape in the icon therefore serves an important functional purpose; it provides context to the otherwise abstract shapes of the icon.

I can't see any benefit from changing the title bar representation to highlight a navigation pane instead - it makes it less clear that the app icon represents an interface, because it abandons the design which users have come to know, instead favouring a novel design that is not widely used or immediately obvious.

To talk more specifically about your proposed design, I also think that attempting to use this navigation pane design is introducing a few problems with the composition of the icon in general. 

The navigation pane in your icon takes up a lot of space on the left side of the icon, meaning the right side of the icon has to accomodate four elements in a limited space:

- the dark blue 'content area' rectangle
- the wrench shape
- the backplate for the 'dev' badge
- the text on the 'dev' badge

This is problematic - it means that, at the smallest sizes, your design ends up rather cramped on the right hand side. This hinders the readability of your icon, because the distance between separate elements of your design is significantly reduced. In fact, in the 16x16 variant of your icon, the wrench shape (which is arguably one of the most important shapes in the icon for recognisability) only gets 1px of room around the top and sides:

![image](https://user-images.githubusercontent.com/11815374/109073010-b7ecac80-76ed-11eb-9064-85f8fd018143.png)

This is bad because oftentimes these icons might be displayed on light backgrounds - the user has to rely on a handful of pixels to define the shape of the entire wrench.

By sticking to the title bar design, the composition of the icon can be greatly improved at these smaller sizes by allowing each element of the icon to have sufficient spacing for easy recognition. There's even enough space to allow the text to stay present at 16x16, which helps because it keeps the design more consistent across multiple icon scales:

![image](https://user-images.githubusercontent.com/11815374/109073230-026e2900-76ee-11eb-89bd-fd29c1c9c09f.png)
![image](https://user-images.githubusercontent.com/11815374/109073722-b1ab0000-76ee-11eb-93f3-ffb5dd212294.png)

The composition here makes more sense - the title bar is free to occupy the upper region of the icon without disturbing the other elements. The 'content area' rectangle then has a lot of space free for it to occupy, giving the wrench shape a much larger background to stand out against, improving readability. Finally, the 'dev' badge can sit unobstructed in the corner, without interfering with the wrench or throwing off the balance of the icon.

Of course, I'm open to change and further discussion, these are just my thoughts right now 🙂

**Comment by marcelwgn** [comment]
Maybe the team can also way in on this, after all they are the ones confronted with this logo the most. @StephenLPeters @ranjeshj feel free to let them know, don't want to mass ping everyone here.

I don't think that the wrench is set in stone here, it existed in the old logo, but I don't think it is something you need to keep up. An issue with the "dev" badge I see is that it suggests there is a non "dev" version (like edge or chrome have).

**Comment by dphfox** [comment]
> An issue with the "dev" badge I see is that it suggests there is a non "dev" version (like edge or chrome have).

Yeah - I initially had reservations about the dev badge for that reason. I don't think it's too big of a deal, but I'd definitely welcome more feedback.

**Comment by mdtauk** [comment]
> Maybe the team can also way in on this, after all they are the ones confronted with this logo the most. @StephenLPeters @ranjeshj feel free to let them know, don't want to mass ping everyone here.
> 
> I don't think that the wrench is set in stone here, it existed in the old logo, but I don't think it is something you need to keep up. An issue with the "dev" badge I see is that it suggests there is a non "dev" version (like edge or chrome have).

The current logo has a hammer, but the wrench is easier to draw geometrically 😜

![image](https://user-images.githubusercontent.com/7389110/109206543-e5922e00-779f-11eb-885e-7516858769c0.png)


This is why I created a Discussion rather than a proposal or PR for this.  Everyone should weigh in with ideas or comments.  Once some kind of consensus is reached, then we can worry about the implementation.

**Comment by marcelwgn** [comment]
> Yeah - I initially had reservations about the dev badge for that reason. I don't think it's too big of a deal, but I'd definitely welcome more feedback.

Problem is that there aren't many options that indicate "testing" or development.

> The current logo has a hammer, but the wrench is easier to draw geometrically 😜

Oh haha yes, totally forgot that haha, you are totally right.

> This is why I created a Discussion rather than a proposal or PR for this. Everyone should weigh in with ideas or comments. Once some kind of consensus is reached, then we can worry about the implementation.

👍

@mdtauk Could you rename the discussion while we are at it? The MUXTestApp is already open source so having a logo is something that shouldn't have to wait on WinUI 3 open source.

**Comment by mdtauk** [comment]
> @mdtauk Could you rename the discussion while we are at it? The MUXTestApp is already open source so having a logo is something that shouldn't have to wait on WinUI 3 open source.
>

Done - I included the bit about opening source because I figured it would not be a priority until then, and the community can get more involved in ongoing work.

**Comment by mdtauk** [comment]
There is this Icon used previously by Xbox for developement/construction
![image](https://user-images.githubusercontent.com/7389110/109207658-4706cc80-77a1-11eb-954d-cec5d44c4d27.png)

![image](https://user-images.githubusercontent.com/7389110/109207795-70275d00-77a1-11eb-8e39-1d3d4804a39c.png)

**Comment by marcelwgn** [comment]
What about a small crane holding the WinUI logo (just uttering random thoughts here). Or how about a wrench working on the WinUI logo?

**Comment by mdtauk** [comment]
The WinUI logo would be pretty small if it has to fit inside a wrench, especially at 12 x 12 or 16 x 16 sizes

**Comment by marcelwgn** [comment]
Maybe not inside but rather underneath it or partially obstructed by the wrench?

**Comment by dphfox** [comment]
Here's some ideas I came up with that involve the WinUI logo - haven't tried cranes or anything yet, but I quite like the look of the three on the right personally;

![image](https://user-images.githubusercontent.com/11815374/109406677-078fda00-7973-11eb-8519-9b754df727dd.png)

They're all designed on a 16x16 grid, so they'll be easy to scale up later and should look crisp at any size:
![image](https://user-images.githubusercontent.com/11815374/109406704-31490100-7973-11eb-995f-6dd81d41c1b5.png)

edit: tested them out on some splash screen mock ups too, to get a better feel for what they'd look like within Windows:

![image](https://user-images.githubusercontent.com/11815374/109407267-56d80980-7977-11eb-8639-addcaf919bf4.png)
